vendor/k8s.io/kubernetes/pkg/client/unversioned/clientcmd/client_config.go

```diff
@@ -90,12 +90,24 @@ func (config *DirectClientConfig) RawConfig() (clientcmdapi.Config, error) {
 
 // ClientConfig implements ClientConfig
 func (config *DirectClientConfig) ClientConfig() (*restclient.Config, error) {
-	if err := config.ConfirmUsable(); err != nil {
+	configAuthInfo, err := config.getAuthInfo()
```

Comment:
add a comment about why we're fetching this first and checking the error (to catch invalid explicit user overrides)